consensus/safety-rules/src/safety_rules.rs

```diff
@@ -67,9 +71,11 @@ impl SafetyRules {
         self.verify_epoch(proposed_block.epoch(), &safety_data)?;
 
         self.verify_qc(proposed_block.quorum_cert())?;
-        proposed_block
-            .validate_signature(&self.epoch_state()?.verifier)
-            .map_err(|error| Error::InvalidProposal(error.to_string()))?;
+        if !self.skip_sig_verify {
+            proposed_block
+                .validate_signature(&self.epoch_state()?.verifier)
+                .map_err(|error| Error::InvalidProposal(error.to_string()))?;
+        }
         proposed_block
```

Comment:
Can also skip this as it is already verified. https://github.com/aptos-labs/aptos-core/blob/zekun/sig/consensus/src/round_manager.rs#L118